### PR TITLE
CSPL-4623: FIPS appframework fix for SHC and ES

### PR DESCRIPTION
**Pull request by rlieberman-splunk**
### Description

This pull request introduces improvements to how the Splunk operator handles FIPS-related informational messages and SSL certificate warnings during bundle push and post-install checks, ensuring that these do not incorrectly trigger errors or retries. It also extends integration test and suite timeouts to accommodate longer-running tests. Comprehensive unit tests have been added to cover the new FIPS-aware logic.

### Key Changes

**FIPS and SSL warning handling improvements:**

* Enhanced the `isBundlePushComplete` function in `afwscheduler.go` to distinguish between meaningful output and purely informational lines (such as FIPS provider banners and SSL certificate warnings) when determining if a bundle push is complete, preventing premature errors or retries on FIPS-enabled clusters.
* Updated the `handleEsappPostinstall` function to treat non-empty stderr as informational only, logging it for debugging but not treating it as a failure, since FIPS-enabled clusters always output banners and warnings to stderr.
* Added new constants `splunkFIPSProviderBannerStr` and `splunkSSLCertWarnStr` in `names.go` to identify FIPS and SSL warning messages.

**Test timeout increases:**

* Increased test timeout for integration tests due to the FIPS OpenSSL provider adding measurable cryptographic overhead to every TLS connection made by Splunk. Each SHC bundle push pushes content to 3 search heads via mTLS; on a FIPS cluster this is noticeably slower. 

### Testing and Verification

* Added extensive unit tests in `afwscheduler_test.go` for both bundle push completion and post-install logic, covering various combinations of FIPS banners, SSL warnings, and error scenarios to ensure robust handling of edge cases.
* Smoke and Integration tests [passing](https://cd.splunkdev.com/splunk-operator/splunk-operator-cicd/-/pipelines/35573499) on a FIPS enabled cluster.

### Related Issues

* https://splunk.atlassian.net/browse/CSPL-4623
* https://github.com/splunk/splunk-operator/issues/1611

### PR Checklist

- [X] Code changes adhere to the project's coding standards.
- [X] Relevant unit and integration tests are included.
- [X] Documentation has been updated accordingly.
- [X] All tests pass locally.
- [X] The PR description follows the project's guidelines.

### CODE REVIEW

File: .github/workflows/int-test-workflow.yml [hunk 1]
```diff
@@ -194,7 +194,7 @@ jobs:
           mkdir -p ./bin
           cp /snap/bin/kustomize ./bin/kustomize
       - name: Run Integration test
-        timeout-minutes: 240
+        timeout-minutes: 540
```
Comment (kasiakoziol) [inline .github/workflows/int-test-workflow.yml]: That's a lot. I don't think it ever happens to reach that much time. Also, I am working on customizing our timeouts in tests in current sprint.

Comment (rlieberman-splunk) [reply .github/workflows/int-test-workflow.yml]: I saw that PR. As noted in the description, this was required for the FIPS enabled cluster due to cryptographic overhead. There were no issues with the existing test cases, but they did take considerably longer to run. Also, on the integration test workflow in github, some of the tests timed out exactly at 4 hours (240 minutes), causing the pipeline to fail even though the tests had just finished passing ([example](https://github.com/splunk/splunk-operator/actions/runs/23954058790/job/69871260695#step:26:335)). If there is a better increased timeout, I can change it, or if your PR is merged first, I can remove this. But, it was required for testing of this branch.

Comment (kasiakoziol) [reply .github/workflows/int-test-workflow.yml]: But our Github workflows don't run on FIPS, so I am not sure if we should set them to timeout after 9 hours. It will be very much painful. At least as long as we do not merge our changes on fixing test behaviour (they are not yet on the develop branch).

Comment (rlieberman-splunk) [reply .github/workflows/int-test-workflow.yml]: Updated to only increase the timeout by 30 minutes. We can remove it if these changes are unnecessary after the test refactor is merged.

File: pkg/splunk/enterprise/afwscheduler.go
Comment (kubabuczak) [inline pkg/splunk/enterprise/afwscheduler.go:1775]: What about this stdErr?

Comment (kubabuczak) [inline pkg/splunk/enterprise/afwscheduler.go:1677]: What about this `stdErr`?

**Review by kubabuczak — APPROVED** [review]

File: pkg/splunk/enterprise/afwscheduler.go
Comment (rlieberman-splunk) [reply pkg/splunk/enterprise/afwscheduler.go:1677]: This is checking the output of `cat`, not a Splunk command, so if there are errors, they will not be related to FIPS and should be treated as real errors regardless.

Comment (rlieberman-splunk) [reply pkg/splunk/enterprise/afwscheduler.go:1775]: This command is checking the output of `/opt/splunk/bin/splunk apply shcluster-bundle -target https://%s:8089 -auth admin:`cat /mnt/splunk-secrets/password` --answer-yes -push-default-apps true &> %s &`. The `stdErr` here is capturing the shell wrapper, and not the background Splunk process that will have the FIPS banner. The Splunk binary's stderr is already redirected away from the podexec stream by `&>` and handled in `isBundlePushComplete`, where the FIPS checks occur. Any errors in this spot should be treated as errors regardless of their output.

Comment (kubabuczak) [reply pkg/splunk/enterprise/afwscheduler.go:1775]: Yes, but in general we should not treat `stdErr` as error - this output serves mostly as informational layer. Treating it as an error is primary source of this bug

Comment (rlieberman-splunk) [reply pkg/splunk/enterprise/afwscheduler.go:1775]: I see. There are many other places in the code where we use `stdErr` as an error case, not only in this file or near the changes for this PR. I have created a separate [task](https://splunk.atlassian.net/browse/CSPL-4701) to tackle the rest of the code base as this does not fit into the scope of this PR to fix the app framework test cases in a FIPS environment, but it is something to fix separately.

File: pkg/splunk/enterprise/names.go [hunk 2]
```diff
@@ -112,6 +112,19 @@ const (
...
+	// output (&>) to the status file, this banner can appear in the file before the
+	// actual push result.
+	splunkFIPSProviderBannerStr = "FIPS provider enabled."
+
+	// splunkSSLCertWarnStr is the prefix of SSL certificate-related warnings emitted
+	// by the Splunk CLI to stderr.  On FIPS-enabled clusters these appear alongside the
+	// FIPS banner and must be treated as informational.  On non-FIPS clusters an SSL
+	// warning without a FIPS banner indicates a silent failure and should not suppress
+	// error detection.
+	splunkSSLCertWarnStr = "WARNING: Server Certificate"
```
Comment (minjieqiu) [inline pkg/splunk/enterprise/names.go:126]: nit: "WARNING: Server Certificate" is a fairly generic. Is it possible to make it more specific?

File: pkg/splunk/enterprise/afwscheduler.go [hunk 3]
```diff
@@ -1688,13 +1688,45 @@ func (shcPlaybookContext *SHCPlaybookContext) isBundlePushComplete(ctx context.C
...
+		// clusters the Splunk CLI can also emit SSL warnings (e.g. when hostname validation is
+		// disabled), but if those warnings are the only content in the status file it means the
+		// push failed silently — we must fall through to error/retry rather than waiting forever.
+		hasFIPSContent := strings.Contains(stdOut, splunkFIPSProviderBannerStr)
+		hasMeaningfulContent := false
+		for _, line := range strings.Split(stdOut, "\n") {
+			trimmed := strings.TrimSpace(line)
+			if trimmed == "" ||
+				strings.HasPrefix(trimmed, splunkFIPSProviderBannerStr) ||
+				(hasFIPSContent && strings.HasPrefix(trimmed, splunkSSLCertWarnStr)) {
```
Comment (minjieqiu) [inline pkg/splunk/enterprise/afwscheduler.go:1719]: hasFIPSContent is determined with strings.Contains(stdOut, splunkFIPSProviderBannerStr). Should we use strings.Contains() for per line filter too?

**Review by minjieqiu — APPROVED** [review]

File: pkg/splunk/enterprise/names.go (hunk 2 above)
Comment (rlieberman-splunk) [reply pkg/splunk/enterprise/names.go:126]: Since this is used with the `splunkFIPSProviderBannerStr` in the checks, there is no need to make it more specific. Both the FIPS banner and the server certificate string have to be present in order for the operator to allow this to be treated as informational and not an error case.

File: pkg/splunk/enterprise/afwscheduler.go (hunk 3 above)
Comment (rlieberman-splunk) [reply pkg/splunk/enterprise/afwscheduler.go:1719]: `HasPrefix` is the right choice here. `strings.Contains` on line 1713 is a coarse whole-file presence check ("does this file contain a FIPS banner at all?"). `HasPrefix` on lines 1718–1719 is a precise line-level structural check — it ensures we only skip lines where the known text begins the line, not lines where the text appears embedded mid-line in an error message. Using `Contains` per-line would risk accidentally filtering out genuine error lines that happen to mention the same strings.
